geonode/mngmt_commands_http/tests.py

```diff
@@ -0,0 +1,135 @@
+#########################################################################
+#
+# Copyright (C) 2021 OSGeo
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#########################################################################
+from rest_framework import status
+from rest_framework.test import APITestCase
+from django.contrib.auth import get_user_model
+from unittest.mock import patch
+from .models import ManagementCommandJob
+from .utils import run_management_command
+
+
+class ManagementCommandsTestCase(APITestCase):
+    def setUp(self):
+        self.resource_url = "/api/v2/management/"
+        self.admin = get_user_model().objects.create_superuser(
+            username="admin",
+            password="admin",
+            email="admin@geonode.org",
+        )
+        self.non_admin_user = get_user_model().objects.create_user(
+            username="some_user",
+            password="some_password",
+            email="some_user@geonode.org",
+        )
+        self.client.force_authenticate(self.admin)
+
+    def test_management_commands_list(self):
+        expected_payload = {
+            "success": True,
+            "error": None,
+            "data": ['ping_mngmt_commands_http'],
+        }
+        response = self.client.get(self.resource_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json(), expected_payload)
+
+    def test_management_commands_list_forbidden(self):
+        self.client.force_authenticate(self.non_admin_user)
+        response = self.client.get(self.resource_url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_management_commands_detail(self):
+        cmd_name = "ping_mngmt_commands_http"
+        resource_url = f"/api/v2/management/{cmd_name}/"
+        response = self.client.get(resource_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = response.json()
+        self.assertTrue(response_json['success'])
+        self.assertIn(cmd_name, response_json['data'])
+
+    def test_management_commands_detail_not_found(self):
+        cmd_name = "some_unavaliable_command"
+        resource_url = f"/api/v2/management/{cmd_name}/"
```

Comment:
If you want to make the resource_url dynamic, you can declare it in the `setUp` as -> `self.resource_url = "/api/v2/management/{}` and then in each function use the `format` to compone it example: `self.client.get(resource_url.format(cmd_name ))`

Reply:
Updated ✅